src/api/context/propagation/text_propagator.rs

```diff
@@ -41,4 +41,9 @@ pub trait HttpTextFormat: Debug {
     /// [`Context`]: ../../struct.Context.html
     /// [`Extractor`]: ../trait.Extractor.html
     fn extract_with_context(&self, cx: &Context, extractor: &dyn api::Extractor) -> Context;
+
+    /// Returns list of fields used by [`HttpTextFormat`]
+    ///
+    /// [`HttpTextFormat`]: ./trait.HttpTextFormat.html
+    fn get_fields(&self) -> Vec<String>;
```

Comment:
It looks like propagator fields are generally static, I wonder if we could avoid the allocations here. Trickier with the composite propagator though. Ideally something like `[&'static str]` would be nice 🤔

Reply:
I actually thought about it. But since it's API, I thought we should allow user tweak it as much as possible. 

Like if user are trying to acquire the header name from some kind of configuration services. You can't really make it a `&'static str`

Reply:
Hm true. Although they would likely configure the propagator once when constructed right? could do `&[String]` maybe. That would work from static, e.g. `static FIELDS: [String; 1] = [TRACEPARENT_HEADER.to_string()];` and could work if composite propagator just stores `fields: Vec<String>`.

In that case could also make this an iterator that you could construct from `&[String]`. Something like:

```rust
/// An iterator over fields of a text propagator
#[derive(Debug)]
pub struct FieldIter<'a>(slice::Iter<'a, String>);

impl<'a> FieldIter<'a> {
    /// Create a new `FieldIter` from a slice of propagator fields
    pub fn new(fields: &'a [String]) -> Self {
        FieldIter(fields.iter())
    }
}

impl<'a> Iterator for FieldIter<'a> {
    type Item = &'a String;
    fn next(&mut self) -> Option<Self::Item> {
        self.0.next()
    }
}
```

Reply:
Yes `&[String]` sounds like a good idea. I would image you will have to put the header names somewhere in your propagator. I am gonna give it a try. Thanks for the advice!